padlockcloud/sender.go

```diff
@@ -27,21 +43,89 @@ type EmailSender struct {
 
 // Attempts to send an email to a given recipient. Through `smpt.SendMail`
 func (sender *EmailSender) Send(rec string, subject string, body string) error {
-	auth := smtp.PlainAuth(
-		"",
-		sender.Config.User,
-		sender.Config.Password,
-		sender.Config.Server,
-	)
-
-	message := fmt.Sprintf("Subject: %s\r\nFrom: Padlock Cloud <%s>\r\n\r\n%s", subject, sender.Config.User, body)
-	return smtp.SendMail(
-		sender.Config.Server+":"+sender.Config.Port,
-		auth,
-		sender.Config.User,
-		[]string{rec},
-		[]byte(message),
-	)
+
+	// Send Email using SMTP Protocol
+	if strings.ToLower(sender.Config.Provider) == "smtp" {
+		//fmt.Println("Sending Email Using SMTP.")
+		log.Print("Using SMTP For Sending Email.")
+		auth := smtp.PlainAuth(
+			"",
+			sender.Config.User,
+			sender.Config.Password,
+			sender.Config.Server,
+		)
+
+		message := fmt.Sprintf("Subject: %s\r\nFrom: Padlock Cloud <%s>\r\n\r\n%s", subject, sender.Config.User, body)
+		return smtp.SendMail(
+			sender.Config.Server+":"+sender.Config.Port,
+			auth,
+			sender.Config.User,
+			[]string{rec},
+			[]byte(message),
+		)
+
+	}
+
+	// Send Email Using AWS SES
+	if strings.ToLower(sender.Config.Provider) == "ses" {
+		//fmt.Println("Sending Email Using AWS SES.")
+		log.Print("Using AWS SES For Sending Email.")
+
+		var sess *session.Session
+		var err error
+
+		// Determine whether the AWS Access Keys are Statically Provided.
+		if sender.Config.AWSAccessKey != "" {
+			log.Print("Using Statically defined credentials is a bad practice. Please see AWS Documentation for more info. https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#id2")
+			sess, err = session.NewSession(&aws.Config{
+				Region:      aws.String(sender.Config.AWSRegion),
+				Credentials: credentials.NewStaticCredentials(sender.Config.AWSAccessKey, sender.Config.AWSSecretKey, ""),
+			})
+		} else {
+			sess, err = session.NewSession(&aws.Config{Region: aws.String(sender.Config.AWSRegion)})
+		}
+
+		if err != nil {
+			log.Print(err)
+		}
+		svc := ses.New(sess)
+
+		params := &ses.SendEmailInput{
+			Destination: &ses.Destination{ // Required
+				ToAddresses: []*string{
+					aws.String(rec), // Required
+				},
+			},
+			Message: &ses.Message{ // Required
+				Body: &ses.Body{ // Required
+					Html: &ses.Content{
+						Data: aws.String(body), // Required
+					},
+					Text: &ses.Content{
+						Data: aws.String(body), // Required
+					},
+				},
+				Subject: &ses.Content{ // Required
+					Data: aws.String("From Padlock Cloud"), // Required
+				},
+			},
+			Source: aws.String(sender.Config.SESFrom), // Required
+		}
+		//fmt.Println(params)
+		_, err = svc.SendEmail(params)
+
+		if err != nil {
+			// Print the error, cast err to awserr.Error to get the Code and
+			// Message from an error.
+			fmt.Println(err.Error())
```

Comment:
Don't print this here, and not with the `fmt` package. If you're returning the error, it should be handled by the caller, i.e. any logging should be done there as well.